internal/config/config.go

```diff
@@ -802,6 +803,46 @@ func (c AzureSEVSNP) EqualTo(old AttestationCfg) (bool, error) {
 	return firmwareSignerCfgEqual && measurementsEqual && bootloaderEqual && teeEqual && snpEqual && microcodeEqual && rootKeyEqual, nil
 }
 
+func (a *AzureSEVSNP) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	aux := &fusedAzureSEVSNP{
+		auxAzureSEVSNP: (*auxAzureSEVSNP)(a),
+	}
+	if err := unmarshal(aux); err != nil {
+		return fmt.Errorf("failed to unmarshal AzureSEVSNP: %w", err)
+	}
+	a = (*AzureSEVSNP)(aux.auxAzureSEVSNP)
+
+	for _, versionType := range []version.Type{version.Bootloader, version.TEE, version.SNP, version.Microcode} {
+		if !convertLatestToNumber(a, versionType, aux) {
+			if !convertStringToUint(a, versionType, aux) {
+				return fmt.Errorf("failed to convert %s version to number", versionType)
+			}
+		}
+	}
+	return nil
+}
+
+func getUintAndStringPtrToVersion(versionType version.Type, c *AzureSEVSNP, aux *fusedAzureSEVSNP) (*uint8, *string) {
```

Comment:
```suggestion
func getUintAndStringPtrToVersion(c *AzureSEVSNP, versionType version.Type, aux *fusedAzureSEVSNP) (*uint8, *string) {
```
nit:  `convertStringToUint` and `convertLatestToNumber` use a different order of function arguments.